testing/sync-test/src/autofill.rs

```diff
@@ -242,6 +256,135 @@ fn test_autofill_addresses_general(c0: &mut TestClient, c1: &mut TestClient) {
+    assert!(credit_card_failures.is_some());
+    assert!(credit_card_failures.unwrap().contains("Crypto Error"));
+
+    // clear records
+    delete_credit_card(&c0.autofill_store, credit_card0)
+        .expect("credit_card0 to be deleted from c0");
+    verify_credit_card_removal(&c0.autofill_store);
+    verify_credit_card_removal(&c1.autofill_store);
+}
+
+fn test_scrub_undecryptable_records_for_remote_replacement(
+    c0: &mut TestClient,
+    c1: &mut TestClient,
+) {
+    log::info!("Add a credit card to client0");
+    let key = create_autofill_key().expect("encryption key created");
+
+    // Add a credit card
+    let credit_card0 = add_credit_card(
+        &c0.autofill_store,
+        UpdatableCreditCardFields {
+            cc_name: "john deer".to_string(),
+            cc_number_enc: encrypt_string(key.clone(), "88888888888888".to_string())
+                .expect("encrypted cc number for credit_card0"),
+            cc_number_last_4: "8888".to_string(),
+            cc_exp_month: 10,
+            cc_exp_year: 2025,
+            cc_type: "mastercard".to_string(),
+        },
+    )
+    .expect("add credit_card0");
+
+    let cc0id = credit_card0.clone().guid;
+
+    log::info!("Verifying credit_card0 on c0");
+    verify_credit_card(&c0.autofill_store, &credit_card0);
+
+    // Sync the first device where the credit card was added
+    log::info!("Syncing client0 -- inital sync");
+    sync_credit_cards(c0, key.clone()).expect("c0 sync to work");
+
+    // Sync the second device
+    log::info!("Syncing client1 -- inital sync");
+    sync_credit_cards(c1, key.clone()).expect("c1 sync to work");
+
+    // Verify that the credit card exists on both devices
+    verify_credit_card(&c0.autofill_store, &credit_card0);
+    verify_credit_card(&c1.autofill_store, &credit_card0);
+
+    log::info!("Scrub and verify credit card");
+    let new_key = create_autofill_key().expect("encryption key created");
+    c0.autofill_store
+        .clone()
+        .scrub_undecryptable_credit_card_data_for_remote_replacement(&new_key)
+        .expect("stored credit card to be scrubbed");
+
+    // Verify that the credit card has been scrubbed
+    let scrubbed_credit_card = c0
+        .autofill_store
+        .clone()
+        .get_credit_card(cc0id.clone())
+        .expect("stored credit card to be retrieved");
+    assert_eq!(scrubbed_credit_card.cc_number_enc, "");
+
+    // Sync the first device after scrubbing
+    log::info!("Syncing client0 -- after scrubbing");
+    sync_credit_cards(c0, key.clone()).expect("c0 sync to work");
+
+    // Sync the second device after scrubbing
+    log::info!("Syncing client1 -- after scrubbing");
+    sync_credit_cards(c1, key.clone()).expect("c1 sync to work");
+
+    let c0_scrubbed_record = c0
+        .autofill_store
+        .clone()
+        .get_credit_card(cc0id.clone())
+        .expect("stored credit card to be retrieved");
+    assert_ne!(c0_scrubbed_record.cc_number_enc, "");
```

Comment:
it would seem marginally better to decrypt the new `cc_number_enc` with the new key and verify it matches the number we expect?